frontend/views/__init__.py

```diff
@@ -422,7 +422,8 @@ def work(request, work_id, action='display'):
         'cover_width': cover_width_number,
         'action': action,
         'formset': formset,
-        'kwform': SubjectSelectForm()
+        'kwform': SubjectSelectForm(),
+        'has_online_book': work.first_epub() != None,
```

Comment:
look at the code for models.Ebook.get_archive_ebf  You want to make sure `ebook.provider in good_providers` and then use archive_ebf as the epub source